### PR TITLE
Reuse gcr.io tokens

### PR DESCRIPTION
**Pull request by dgageot**
Also auto-configure gcloud as a credential helper
when that makes sense.

Signed-off-by: David Gageot <david@gageot.net>

### CODE REVIEW

File: pkg/skaffold/docker/authenticators.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+var authenticators = Authenticators{}
+
+// Authenticators stores an authenticator per registry.
+type Authenticators struct {
+	byRegistry map[string]*lockedAuthenticator
+	lock       sync.Mutex
+}
+
+// For retrieves the authentiator for a given image reference.
```
Comment (jonjohnsonjr) [inline pkg/skaffold/docker/authenticators.go]: This reuse is something we could reasonably do in `google.Keychain`

Comment (dgageot) [reply pkg/skaffold/docker/authenticators.go]: Go for it! It'll make it easier for Skaffold

**Comment by jonjohnsonjr** [comment]
@dgageot what do you think about https://github.com/google/go-containerregistry/pull/603

**Comment by dgageot** [comment]
I'm going to merge this one. When go-containerregistry has the code required to do the caching, I'll change Skaffold.

**Comment by jonjohnsonjr** [comment]
It's merged now :smile:

**Pull request merged**